test/framework/cluster_proxy.go

```diff
@@ -247,6 +253,27 @@ func (p *clusterProxy) Apply(ctx context.Context, resources []byte, args ...stri
 	return exec.KubectlApply(ctx, p.kubeconfigPath, resources, args...)
 }
 
+// Create creates using the controller-runtime client.
+func (p *clusterProxy) Create(ctx context.Context, resources []byte) error {
+	Expect(ctx).NotTo(BeNil(), "ctx is required for Create")
+	Expect(resources).NotTo(BeNil(), "resources is required for Create")
+
+	var retErrs []error
+	objs, err := yaml.ToUnstructured(resources)
+	if err != nil {
+		return err
+	}
+	for _, o := range objs {
+		if err := p.GetClient().Create(ctx, &o); err != nil {
```

Comment:
```suggestion
	for o := range objs {
		if err := p.GetClient().Create(ctx, &objs[o]); err != nil {
```

or let's reassign the iteration var
```
for _, o := range objs { 
   o := o
   if err := p.GetClient().Create(ctx, &o); err != nil {
```